### DISCUSSION TITLE
How did you reverse engineer this?

### DISCUSSION THREAD

**Discussion by dbedrenko** [emu-russia/psxcpu#47] [Q&A] [answered]
Hello, this is extraordinary! I wonder how does a person reverse engineer a CPU? Is there anywhere I can read how you did this? I am very curious about this.

I understand that some documents that explain the architecture are available. But did you have to decap the chip, and look at the Integrated Circuit under a microscope? How could you comprehend anything coherent out of that?! This is amazing.

**Comment by ogamespec** [comment, marked as answer]
Yes, you have to take a picture first, of course. I described the whole process a little bit here: https://github.com/emu-russia/dmgcpu/blob/main/methods.md

**Reply by dbedrenko** [reply]
Thank you! I'm learning so much, I love it. It must've taken so much work
